state-reconstruct-fetcher/src/perf_metric.rs

```diff
@@ -0,0 +1,31 @@
+use tokio::time::Duration;
+
+#[derive(Default)]
+pub struct PerfMetric {
+    total: Duration,
+    count: u32,
+}
+
+impl PerfMetric {
+    pub fn add(&mut self, duration: Duration) -> u32 {
+        self.total += duration;
+        self.count += 1;
+        self.count
+    }
+
+    pub fn renew(&mut self) -> String {
+        let old = self.format();
+        self.total = Duration::default();
+        self.count = 0;
+        old
+    }
+
+    pub fn format(&self) -> String {
+        if self.count == 0 {
+            String::from("-")
+        } else {
+            let duration = self.total / self.count;
+            format!("{:?}", duration)
+        }
+    }
```

Comment:
I would implement this under [`std::fmt::Display`](https://doc.rust-lang.org/std/fmt/trait.Display.html) instead to gain some nice benefits. Like so:
```rust
impl Display for PerfMetric {
    fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), Error>
        if self.count == 0 {
            write(f, "-")
        } else {
            let duration = self.total / self.count;
            write!(f, "{:?}", duration)
        }
    }
}
```

Reply:
OK. BTW, do you happen to know how to format floating-point numbers (i.e. Duration milliseconds) with optional scientific notation?